src/factory.js

```diff
@@ -62,21 +62,52 @@ BlockFactory.oldDir = null;
 
 /*
  * The starting XML for the Block Factory main workspace. Contains the
- * unmovable, undeletable factory_base block.
- */
-BlockFactory.STARTER_BLOCK_XML_TEXT = '<xml><block type="factory_base" ' +
-    'deletable="false" movable="false">' +
-    '<value name="TOOLTIP">' +
-    '<block type="text" deletable="false" movable="false">' +
-    '<field name="TEXT"></field></block></value>' +
-    '<value name="HELPURL">' +
-    '<block type="text" deletable="false" movable="false">' +
-    '<field name="TEXT"></field></block></value>' +
-    '<value name="COLOUR">' +
-    '<block type="colour_hue">' +
-    '<mutation colour="#5b67a5"></mutation>' +
-    '<field name="HUE">230</field>' +
-    '</block></value></block></xml>';
+ * unmovable, undeletable factory_base block. Allows user input to change XML
+ * starter block.
+ * @param {string} inputType Type of input (statement, value, dummy).
+ * @param {string} blockStarterText Starter text to place on block, given by user (optional).
+ * @param {string} blockTypeName Name of block, given by user.
+*/
+BlockFactory.insertXMLInputs = function(inputType, blockStarterText, blockTypeName) {
```

Comment:
consider naming this 'buildStartXml' or something similar. insert implies it's adding it to something that exists instead of returning a new thing. Also, note that in camelCase Xml is treated as a word instead of being kept all caps.